From 1ff0f62a06ca1e3f6679ca017d605e5d906e0e63 Mon Sep 17 00:00:00 2001
From: David Spickett <david.spickett@linaro.org>
Date: Fri, 11 Feb 2022 11:09:28 +0000
Subject: [PATCH] [compiler-rt][xray] Disable fdr-reinit test on AArch64

We run bots on a shared machine and under high load
this test sometimes segfaults.

https://lab.llvm.org/buildbot/#/builders/185/builds/1368

==1952234==XRay FDR init successful.
==1952234==XRay FDR: Not flushing to file, 'no_file_flush=true'.
<...>fdr-reinit.cpp.script: line 4: 1952234 Segmentation fault
XRAY_OPTIONS="verbosity=1" <...>/fdr-reinit.cpp.tmp

Looking at the printed output I think it's happening at:
// Finally, we should signal the sibling thread to stop.
keep_going.clear(std::memory_order_release);

Disabling the test while I try to reproduce.
---
 compiler-rt/test/xray/TestCases/Posix/fdr-reinit.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/compiler-rt/test/xray/TestCases/Posix/fdr-reinit.cpp b/compiler-rt/test/xray/TestCases/Posix/fdr-reinit.cpp
index a8f2608f02be..42e7fa24d54e 100644
--- a/compiler-rt/test/xray/TestCases/Posix/fdr-reinit.cpp
+++ b/compiler-rt/test/xray/TestCases/Posix/fdr-reinit.cpp
@@ -1,3 +1,5 @@
+// This will sometimes segfault on the AArch64 bots
+// UNSUPPORTED: aarch64
 // RUN: %clangxx_xray -g -std=c++11 %s -o %t
 // RUN: rm xray-log.fdr-reinit* || true
 // RUN: XRAY_OPTIONS="verbosity=1" %run %t
-- 
2.39.3

